src/triage/component/audition/thresholding.py

```diff
@@ -21,6 +21,66 @@ def _of_metric(df, metric_filter):
     ]
 
 
+class ModelGroupChecker(object):
+    def __init__(
+        self,
+        train_end_times,
+        initial_model_group_ids,
+        models_table,
+        db_engine
+    ):
+        """Before create the distance table, we want to check if the model_group_ids
+        and train_end_times passed in are reasonable to prevent incomparable model groups
+        from being populated to the distance table.
+
+        Args:
+            train_end_times (list) The set of train end times to consider during iteration
+            initial_model_group_ids (list) The initial list of model group ids to narrow down
+            models_table (string) The name of the results schema
+            db_engine (sqlalchemy.engine) A database engine with access to a results schema
+                of a completed modeling run
+        """
+        self.train_end_times = train_end_times
+        self._initial_model_group_ids = initial_model_group_ids
+        self.models_table = models_table
+        self.db_engine = db_engine
+
+    def _check_list_of_string(self, obj):
+        return bool(obj) and all(isinstance(element, str) for element in obj)
```

Comment:
Instances of `str` also satisfy this test:

    >>> all(isinstance(el, str) for el in 'foo')
    True

Moreover, you only use this internal method once.

It might make more sense to check type upon construction:

    if isinstance(train_end_times, str) or not hasattr(train_end_times, '__iter__'):
        raise TypeError("hey! …")

…and you could even convert at that point as well, though you don't need to; you could alternatively do so lazily, as you are already.

    end_times_sql = "ARRAY[{}]".format(
        ', '.join(
            "'{}'".format(
                end_time.strftime('%Y -%m-%d') if isinstance(end_time, datetime) else end_time
            ) for end_time in train_end_times
        )
    )